src/Microsoft.OData.Core/HttpUtils.cs

```diff
@@ -433,18 +431,22 @@ internal static string ReadTokenOrQuotedStringValue(string headerName, string he
                 }
             }
 
-            char currentChar = default(char);
+            // if not a quoted string, reading it as Token
+            if (!isQuotedString)
+            {
+                return ReadTokenValue(headerName, headerText, ref textIndex, createException);

```

Comment:
The previous behaviour looks different from the previous version where we throw an exception, is this adding a new feature or support for new behaviour?